src/libraries/Common/src/System/Security/Cryptography/PqcBlobHelpers.cs

```diff
@@ -0,0 +1,178 @@
+                    break;
+                case KeyBlobMagicNumber.BCRYPT_MLDSA_PRIVATE_SEED_MAGIC:
+                    blobType = Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PRIVATE_SEED_BLOB;
+                    break;
+                default:
+                    Debug.Fail("Unknown blob type.");
+                    throw new CryptographicException();
+            }
+
+            return data;
+        }
+
+        private static TResult EncodePQDsaBlob<TResult>(
+            KeyBlobMagicNumber magic,
+            ReadOnlySpan<char> parameterSet,
+            ReadOnlySpan<byte> data,
+            EncodeBlobFunc<TResult> callback)
+        {
+            int blobHeaderSize = Unsafe.SizeOf<BCRYPT_PQDSA_KEY_BLOB>();
+            int parameterSetLengthWithNullTerminator = sizeof(char) * (parameterSet.Length + 1);
+
+            int blobSize =
+                blobHeaderSize +
+                parameterSetLengthWithNullTerminator +  // Parameter set, '\0' terminated
+                data.Length;                            // Key
+
+            byte[] rented = CryptoPool.Rent(blobSize);
+            Span<byte> blobBytes = rented.AsSpan(0, blobSize);
+
+            try
+            {
+                int index = 0;
+
+                // Write header
+                ref BCRYPT_PQDSA_KEY_BLOB blobHeader = ref MemoryMarshal.AsRef<BCRYPT_PQDSA_KEY_BLOB>(blobBytes);
+                blobHeader.Magic = magic;
+                blobHeader.cbParameterSet = parameterSetLengthWithNullTerminator;
+                blobHeader.cbKey = data.Length;
+                index += blobHeaderSize;
+
+                // Write parameter set
+                Span<char> blobBodyChars = MemoryMarshal.Cast<byte, char>(blobBytes.Slice(index));
+                parameterSet.CopyTo(blobBodyChars);
+                blobBodyChars[parameterSet.Length] = '\0';
+                index += parameterSetLengthWithNullTerminator;
+
+                // Write key
+                data.CopyTo(blobBytes.Slice(index));
+                index += data.Length;
+
+                Debug.Assert(index == blobBytes.Length);
+                return callback(blobBytes);
+            }
+            finally
+            {
+                CryptoPool.Return(rented);
+            }
+        }
+
+        private static ReadOnlySpan<byte> DecodePQDsaBlob(
+            ReadOnlySpan<byte> blobBytes,
+            out KeyBlobMagicNumber magic,
+            out ReadOnlySpan<char> parameterSet)
+        {
+            int index = 0;
+
+            ref readonly BCRYPT_PQDSA_KEY_BLOB blob = ref MemoryMarshal.AsRef<BCRYPT_PQDSA_KEY_BLOB>(blobBytes);
+            magic = blob.Magic;
+            int parameterSetLength = blob.cbParameterSet - 2; // Null terminator char, '\0'
+            int keyLength = blob.cbKey;
+            index += Unsafe.SizeOf<BCRYPT_PQDSA_KEY_BLOB>();
+
+            parameterSet = MemoryMarshal.Cast<byte, char>(blobBytes.Slice(index, parameterSetLength));
```

Comment:
I don't think we should use `MemoryMarshal` here, one because it's not needed, and two I don't think this will work when you try to light it up for Microsoft.Bcl.Cryptography. All of this marshal logic is going to need to work on .NET Framework.

Reply:
Microsoft.Bcl.Cryptography already references System.Memory where `MemoryMarshal.Cast<,>` is available for netstandard2.0 (and `AsRef` can be built as `foo.Cast<byte,T>()[0]`). This feels more natural to me rather than unsafe and pinning (maybe because I don't do much interop).